json/from_json.mbt

```diff
@@ -137,7 +137,10 @@ pub impl[X : FromJson] FromJson for Array[X] with from_json(json, path) {
       let res : Array[X] = Array::new(capacity=a.length())
       let idx = Index(path, index=0)
       for i = 0; i < a.length(); i = i + 1 {
-        idx.update_index_inplace(i)
+        // TODO: it should be assert here
+        if idx is (Index(_) as idx) {
+          idx.index = i
+        }
```

Comment:
FTR: in theory, the compiler know idx is a subtype, we don't need such assertion